src/main/webapp/app/modeling-assessment/modeling-assessment.component.ts

```diff
@@ -210,25 +205,23 @@ export class ModelingAssessmentComponent implements AfterViewInit, OnDestroy, On
         }
     }
 
-    private updateHighlightedElements(newElementIDs: Set<string>) {
-        if (!newElementIDs) {
-            newElementIDs = new Set<string>();
+    /**
+     * Sets the corresponding highlight color in the apollon model of all elements contained in the given element map.
+     *
+     * @param newElements a map of elementIds -> highlight color
+     */
+    private updateHighlightedElements(newElements: Map<string, string>) {
+        if (!newElements) {
+            newElements = new Map<string, string>();
```

Comment:
So this is mutating the input `newElements` which is in both cases when this method is called `this.highlightedElements`.
This can lead to suprising behavior when a class property is changed.

I would propose to stick to a certain style here:
- Imperative: Directly use `this.highlightedElements` in the method
- Functional: Don't alter newElements in the method, but make sure it is always initiliazed to an empty `Map` (this would be prefered by me, as it helps testing)